src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -684,12 +684,13 @@ deparse_shard_index_statement(IndexStmt *origStmt, Oid distrelid, int64 shardid,
 							  StringInfo buffer)
 {
 	IndexStmt *indexStmt = copyObject(origStmt); /* copy to avoid modifications */
-	char *relationName = indexStmt->relation->relname;
-	char *indexName = indexStmt->idxname;
 
 	/* extend relation and index name using shard identifier */
-	AppendShardIdToName(&relationName, shardid);
-	AppendShardIdToName(&indexName, shardid);
+	AppendShardIdToName(&(indexStmt->relation->relname), shardid);
```

Comment:
I cannot follow how this solves the issue. Why don't we use your regular steps?
```C
char *shardName = pstrdup(indexStmt->relation->relname);
AppendShardIdToName(&relationName, shardid);

char *shardIndexName = pstrdup(indexStmt-> idxname);
AppendShardIdToName(&shardIndexName, shardid);
```

Reply:
Ok, sorry. This is slightly different that what  we typically do as we call `transformIndexStmt` on `indexStmt`.

Let  me think a bit more

Reply:
As we first copy `indexStmt`, I thought we might want to replace the strings in `indexStmt` as well.